stdlib/public/Concurrency/TaskCancellation.swift

```diff
@@ -22,27 +22,51 @@ import Swift
 /// and reacting to it in that the cancellation handler is _always_ and
 /// _immediately_ invoked when the task is canceled. For example, even if the
 /// operation is running code that never checks for cancellation, a cancellation
-/// handler still runs and provides a chance to run some cleanup code.
+/// handler still runs and provides a chance to run some cleanup code:
+///
+/// ```
+/// await withTaskCancellationHandler {
+///   var sum = 0
+///   while condition {
+///     sum += 1
+///   }
+///   return sum
+/// } onCancel: {
+///   // this onCancel closure may execute concurrently with the operation!
+///   condition.cancel()
+/// }
+/// ```
+///
+/// ### Execution order and semantics
+/// The `operation` closure is always invoked, even when the
+/// `withTaskCancellationHandler` method is called from an already cancelled
+/// task.
+///
+/// When `withTaskCancellationHandler` is used in a task that has already been
+/// cancelled, the cancellation handler will be executed
+/// immediately before the `operation` closure gets to execute.
+///
+/// This allows the cancellation handler to set some external "cancelled" flag
+/// that the operation may be *atomically* checking for in order to avoid
+/// performing any actual work once the operation gets to run.
+///
+/// The `operation` executes on the calling execution context and does not
+/// suspend or change execution context unless code contained within the closure
+/// does so. In other words, the potential suspension point of the
+/// `withTaskCancellationHandler` does not ever suspend by itself before
+/// executing the `operation`.
+///
+/// If cancellation occurs while the operation is running, the cancellation
+/// `handler` will execute *concurrently* with the `operation`.
```

Comment:
```suggestion
/// The `operation` closure executes on the calling execution context, and doesn't
/// suspend or change execution context unless code contained within the closure
/// does so. In other words, the potential suspension point of the
/// `withTaskCancellationHandler(operation:onCancel:)` never suspends by itself before
/// executing the operation.
///
/// If cancellation occurs while the operation is running, the cancellation
/// handler executes *concurrently* with the operation.
```

Avoid future tense, use contractions, use code voice only when referring to a symbol in code.